lib/puppet/provider/dns_record/dynect.rb

```diff
@@ -0,0 +1,149 @@
+
+# Author:: Charles Dunbar <charles@puppetlabs.com>
+# Type Name:: dns_record
+# Provider:: dynect
+#
+# Copyright 2015, Puppet Labs
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require 'json'
+
+begin
+    require "rest_client"
+rescue LoadError
+    raise Puppet::Error, "Missing gem 'rest-client'"
+end
+
+Puppet::Type.type(:dns_record).provide(:dynect) do
+
+  desc "Manage DynECT records."
+
+  def get_token
```

Comment:
I'd recommend replacing this with an `initialize` method so that it is ran only one, not for every call to the `exists?` method.

https://github.com/puppetlabs/puppet/blob/master/lib/puppet/provider/package/windows/package.rb#L84